src/main/scala/com/databricks/spark/xml/parsers/StaxXmlParser.scala

```diff
@@ -159,15 +173,12 @@ private[xml] object StaxXmlParser extends Serializable {
     while (!shouldStop) {
       parser.nextEvent match {
         case e: StartElement =>
```

Comment:
For Map and Arrow, it's not going to fill the partial results. It's matched to Apache Spark's JSON and actually quite complicated to support.